addons/mail/static/src/models/thread/thread.js

```diff
@@ -1512,13 +1519,13 @@ function factory(dependencies) {
             compute: '_computeLastCurrentPartnerMessageSeenByEveryone',
             dependencies: [
                 'partnerSeenInfos',
-                'orderedMessages',
+                'orderedNonTransientMessages',
                 'messagingCurrentPartner',
             ],
         }),
         lastMessage: many2one('mail.message', {
```

Comment:
`lastMessage` is not actually last message, but last non transient message. Should also be renamed to be more specific (e.g. `lastNonTransientMessage`).

Reply:
Ok for renames if we do it quickly, but renaming fields in stable is not something we should do.

Reply:
We don't have the same problems as with python fields. Rename is also a fix, because these fields were intentionally used just for genuine messages.

Reply:
We don't have DB problems but fields should be considered the same as public attributes / methods (which means no API change) to avoid breaking custom fixes/code based on them (either on our prod or by clients).

Reply:
I think the best when it's not an issue is just to create new fields. Here it wouldn't hurt to keep the unused field and add a new one.

Reply:
> I think the best when it's not an issue is just to create new fields. Here it wouldn't hurt to keep the unused field and add a new one.

Yes: `orderedMessages` (keep old) + `orderedNonTransientMessages` (new field) + replace uses of `orderedMessages` by `orderedNonTransientMessages` when needed for fix. That's the step if we want to keep `orderedMessages` being as all ordered messages.

Previous discussion resulted in just renaming `orderedMessages` to `orderedNonTransientMessages`, so a change of meaning of a field may also imply a change of meaning of fields that rely on this, which is the case here with `lastMessage`.

Reply:
The last message from @alexkuhn is a summary. Keep all existing fields. Add new fields for non transient. Change existing field usage to non transient fields in the code where it is appropriate. If computed fields (such as lastMessage) directly depend on a new transient field, create a new computed field for it too.

The rule is we need fields excluding transient almost everywhere, but we are in stable so we don't want to change the meaning of existing fields when possible, so we create new fields instead.

Reply:
> Can one of you summarize what should be modified because I'm not sure I understood all of this discussion (or maybe just force push if little changes) so that we can move forward to the merge ?

My initial comment was about `orderedMessages` not matching its name with changes in PR. Decision was to rename to `orderedNonTransientMessages`. With this change, `lastMessage` field also doesn't match its name, and should be renamed to `lastNonTransientMessage`.

SEB highlights field rename must be discouraged in stable, so another way to fix the issue while not renaming fields is to keep old `orderedMessages` without filter on transient messages and introduce another computed field `orderedNonTransientMessages`.